test/ffc-rules-engine.test.js

```diff
@@ -176,42 +137,120 @@ describe('Rule: Claimed perimeter <= actual perimeter', () => {
         expect(returnedFactsResults[0]).toBe(75)
       })
-      .run({ parcel, claimedPerimeter: 0 })
+      .run({ parcel, quantity: 0 })
   })
 })
 
 describe('Rule: Claimed perimeter <= perimeter adjusted to take into account perimeter features', () => {
   test('Passes when there are no perimeter features and claimed perimeter less than perimeter', async () => {
-    const parcel = {
-      parcelRef: 'PR123',
-      perimeter: 75,
-      perimeterFeatures: [],
-      previousActions: [],
-      sssi: false
-    }
+    const parcel = getParcelWithDefaults({ totalPerimeter: 75 })
+    const result = await runEngine(
+      [rules.adjustedPerimeter],
+      { parcel, quantity: 40 }
+    )
+
+    expect(result.events.length).toBe(1)
+    expect(result.events[0].type).toBe('withinAdjustedPerimeter')
+  })
+  test('Passes when claimed perimeter is less than adjusted perimeter', async () => {
+    const parcel = getParcelWithDefaults({
+      totalPerimeter: 75,
+      perimeterFeatures: [{ type: 'lake', length: 15 }]
+    })
     const result = await runEngine(
       [rules.adjustedPerimeter],
-      { parcel, claimedPerimeter: 40 }
+      { parcel, quantity: 40 }
     )
 
     expect(result.events.length).toBe(1)
     expect(result.events[0].type).toBe('withinAdjustedPerimeter')
   })
+  test('Passes when claimed perimeter is equal to adjusted perimeter', async () => {
+    const parcel = getParcelWithDefaults({
+      totalPerimeter: 75,
+      perimeterFeatures: [{ type: 'lake', length: 15 }]
+    })
+    const result = await runEngine(
+      [rules.adjustedPerimeter],
+      { parcel, quantity: 60 }
+    )
+
+    expect(result.events.length).toBe(1)
+    expect(result.events[0].type).toBe('withinAdjustedPerimeter')
+  })
+  test('Fails when claimed perimeter is greater than adjusted perimeter', async () => {
+    const parcel = getParcelWithDefaults({
+      totalPerimeter: 75,
+      perimeterFeatures: [{ type: 'lake', length: 15 }]
+    })
+    const result = await runEngine(
+      [rules.adjustedPerimeter],
+      { parcel, quantity: 61 }
+    )
+
+    expect(result.events.length).toBe(0)
+  })
+  test('Returns adjusted perimeter in result', async () => {
+    // Note: this test is demonstrating how to return a calculated maximum value for a perimeter
+    // It is not testing any of our code
+    const parcel = getParcelWithDefaults({
+      totalPerimeter: 75,
+      perimeterFeatures: [{ type: 'lake', length: 15 }]
+    })
+    expect.assertions(2)
+    await getEngine([rules.adjustedPerimeter])
+      .on('success', (event, almanac, ruleResult) => {
+        const returnedFactsResults = ruleResult.conditions.all.map((c) => c.factResult)
+        expect(returnedFactsResults.length).toBe(1)
+        expect(returnedFactsResults[0]).toBe(60)
+      })
+      .run({ parcel, quantity: 0 })
+  })
+})
+
+describe('Rule: Claimed perimeter <= perimeter (within accepted tolerance)', () => {
+  const parcel = getParcelWithDefaults({})
```

Comment:
I think we should specify the parcel perimeter here to ensure that the perimeter with tolerance tests are valid regardless of the default parcel data.